Lesson15/deniz/script.js

```diff
@@ -0,0 +1,146 @@
+const updatedArray = ex8Array.map(word => word[0].toUpperCase()+ word.slice(1));
+console.log('result:', updatedArray.join(' '));
+console.log('------------------------');
+// 9. Use the `join` method to join an array of strings into a single string and print it to the console.
+const arrayExercise9 = ['An', 'array', 'of', 'strings'];
+
+console.log("egzersiz 9:", arrayExercise9.join());
+console.log("egzersiz 9:", arrayExercise9.join(' '));
+console.log("egzersiz 9:", arrayExercise9.join(' Exercise '));
+
+// MDN WEB DOC ÖRNEK
+// const elements = ['Fire', 'Air', 'Water'];
+
+// console.log(elements.join());
+// // Expected output: "Fire,Air,Water"
+
+// console.log(elements.join(''));
+// // Expected output: "FireAirWater"
+
+// console.log(elements.join('-'));
+// // Expected output: "Fire-Air-Water"
+
+console.log('------------------------');
+// 10. Use the `replace` method to replace "can help" with "help" in the string stringExercise10 and print the result to the console.
+const stringExercise10 =
+	'Software design patterns can help simplify complex systems.';
+
+console.log("egzersiz 10:", stringExercise10.replace('can help', 'help'));
+console.log('------------------------');
+// 11. Use the `startsWith` method to check if strings below starts with a price, and print
+// the result to the console.
+const string1Exercise11 = '$9.99 - Sale! 50% off all items in store';
+const string2Exercise11 =
+	'Bundle deal: Save $50 when you buy two products together';
+const priceSymbol = '$';
+
+console.log("Egzersiz 11.1:", string1Exercise11.startsWith(priceSymbol));
+console.log("Egzerssiz 11.2:", string2Exercise11.startsWith(priceSymbol));
+console.log('------------------------');
+// 12. Use the `endsWith` method to check if a string ends with a "!" (exclamation mark), and print the result to the console.
+const string1Exercise12 = '$9.99 - Sale! 50% off all items in store';
+const string2Exercise12 =
+	'Bundle deal: Save $50 when you buy two products together!';
+
+console.log('Exercise 12.1: ', string1Exercise12.endsWith("!"));
+console.log('Exercise 12.2: ', string2Exercise12.endsWith("!"));
+console.log('------------------------');
+// 13. Use the `includes` method to check if a string includes a "%" character and print
+// the result to the console.
+const string1Exercise13 = '$9.99 - Sale! 50% off all items in store';
+const string2Exercise13 =
+	'Bundle deal: Save $50 when you buy two products together!';
+const percentSymbol = '%';
+
+console.log('Egzersiz 13:', string1Exercise13.includes("%"));
+console.log('Egzersiz 13:', string2Exercise13.includes("%"));
+console.log('------------------------');
+// 14. Use the `substring` method to extract the word "Databases" from string stringExercise14
+// based on two indices and print it to the console.
+const stringExercise14 =
+	'Databases are an essential component of many applications.';
+
+console.log('Egezrsiz 14: ', stringExercise14.substring(0, 8));
```

Comment:
In real life, we usually work with dynamic data so hardcoding indexes like `(0, 8)` would probably not work. In that case, you'd need to solve this problem by calculating the indexes instead:
```

const startIndex = stringExercise14.indexOf('Databases');
const endIndex = startIndex + 'Databases'.length;
console.log('Egezrsiz 14: ', stringExercise14.substring(startIndex, endIndex));
```